lib/surface/compiler.ex

```diff
@@ -676,7 +676,8 @@ defmodule Surface.Compiler do
             `slot #{slot_name}, props: [..., #{inspect(prop)}]`\
             """
 
-            IOHelper.compile_error(message, prop_meta.file, prop_meta.line)
+            line = prop_meta.line + prop_meta.line_offset - 1
```

Comment:
@lnr0626 although this fixes the problem, I'm not sure it's the best way to handle it. Isn't `prop_meta.line` already supposed to be the line since the beginning of the file?

Currently, using `live_render` in tests hides similar issues as it creates a dynamic view in a single line. I don't remember exactly why it was implemented like that but it was probably because before the new compiler, we didn't have much control over the lines. I think now we'll be able to try a different approach. We can do that on a separate PR.

Reply:
Sorry for the lack of context. Without this change, the error points to the wrong line. Instead of line 2 in the template, it points to line 2 in the whole file:

![image](https://user-images.githubusercontent.com/457237/94991541-3e93c800-055a-11eb-8dba-837f7d7384c2.png)
 
As I said, I'm not sure this is the right way the fix it because we need to make sure the value of `prop_meta.line` is being calculated correctly. Otherwise, we might be just hiding an existing bug.

Reply:
good catch - there was a bug with let and debug where it wasn't actually calculating the line numbers correctly

Reply:
It's just great you could find the root of the problem. Thanks!

I've been playing around with some examples, trying to simulate some common mistakes the user can make so we can make sure we have helpful error/warning messages for most situations.